libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/route/line/api/MapboxRouteLineView.kt

```diff
@@ -142,14 +142,32 @@ class MapboxRouteLineView(options: MapboxRouteLineOptions) {
      * the layers if they have not yet been initialized. If you have a use case for initializing
      * the layers in advance of any API calls this method may be used.
      *
+     * If the layers already exist they will be removed and re-initialized with the options provided.
+     *
      * Each [Layer] added to the map by this class is a persistent layer - it will survive style changes.
      * This means that if the data has not changed, it does not have to be manually redrawn after a style change.
      * See [Style.addPersistentStyleLayer].
      *
      * @param style a valid [Style] instance
      */
     fun initializeLayers(style: Style) {
-        MapboxRouteLineUtils.initializeLayers(style, options)
+        resetLayers(style)
+    }
+
+    /**
+     * Updates the [MapboxRouteLineOptions] used for the route line related layers and initializes the layers.
+     * If the layers already exist they will be removed and re-initialized with the options provided.
+     *
+     * Each [Layer] added to the map by this class is a persistent layer - it will survive style changes.
+     * This means that if the data has not changed, it does not have to be manually redrawn after a style change.
+     * See [Style.addPersistentStyleLayer].
+     *
+     * @param style a valid [Style] instance
+     * @param options used for the route line related layers.
+     */
+    fun initializeLayers(style: Style, options: MapboxRouteLineOptions) {
```

Comment:
I think this is a different case.  In `RouteArrowView` the options weren't mutable but in `MapboxRouteLineView` they are. Also the `initializeLayers` method is used by developers.  Providing this additional method is a convenience over instantiating a new instance just to turn around and call the current `initializeLayers` method.  From a usability point of view this makes the most IMO. 

There's no relationship between this class and `RouteArrowView` their interfaces shouldn't be artificially coupled.

Reply:
> in `MapboxRouteLineView` they are

The options in the `MapboxRouteLineView` have been exposed unintentionally, as we discussed in the previous PRs and as the docs say.

> the `initializeLayers` method is used by developers

The method is used but it's not required to be used, which makes a significant difference. The reset of the layers has to happen once for each fresh instance of the `MapboxRouteLineView` either way (for developers that do not use the `initializeLayers` method), so I'd strongly prefer not to expose a new API and rely on that rebuild on first render instead.

> There's no relationship between this class and `RouteArrowView` their interfaces shouldn't be artificially coupled.

They do follow a similar paradigm though, so learning from one do transition to the other.

Reply:
> The reset of the layers has to happen once for each fresh instance of the MapboxRouteLineView...

Why? This is a new behavior.  

1. Currently instantiating a new `MapboxRouteLineView` will not automatically result in re-initializing the layers - _No change_ in this PR
2. Currently every render call will initialize the layers if they **are not** present in the map style. - _No change_ in this PR
3. Currently a call to initialize the layers will only execute if the layers **are not** in the provided style - _Changes_ in this PR, _every_ call to initialize the layers will execute with the options referenced by `MapboxRouteLineView`.

When I think about a developer using a dependency injection framework it's a lot easier for me to call initializeLayers(style, options) or initializeLayers(style) when I know the options have changed than it is to create a new instance of `MapboxRouteLineView` and replace it in the dependency graph just so the layers get initialized.  

Calling initializeLayers implies a side effect and is more intentional and explicit than instantiating a new `MapboxRouteLineView` to initialize the layers.

Reply:
> Currently instantiating a new MapboxRouteLineView will not automatically result in re-initializing the layers

Options are provided through the constructor, if I as a developer provide new options to my new instance of the `MapboxRouteLineView` and these options are simply ignored, that's clearly a bug that we should fix. Could we please do that?

Reply:
I made some adjustments but I don't think this is the correct behavior.  Initializing of the layers has never been tied to the instantiation of the class. I has been tied to the render methods.  IMO the the re-initializing of the layers with updated options should be explicit rather than hidden in instantiation.